app/models/portfolio.rb

```diff
@@ -0,0 +1,26 @@
+class Portfolio < ApplicationRecord
+  belongs_to :user
+  belongs_to :address
+
+  def self.sync_addresses(user, address_hashes)
+    transaction do
+      address_hashes.each do |address_hash|
```

Comment:
If there has an 1k count address_hashes, the transaction will be slow.I recommend to use `upsert_all` to decrease sql query.